server/src/main/java/org/elasticsearch/reservedstate/service/FileSettingsService.java

```diff
@@ -0,0 +1,309 @@
+        }
+
+        this.watcherThreadLatch = new CountDownLatch(1);
+
+        new Thread(() -> {
+            try {
+                logger.info("file settings service up and running [tid={}]", Thread.currentThread().getId());
+
+                WatchKey key;
+                while ((key = watchService.take()) != null) {
+                    /**
+                     * Reading and interpreting watch service events can vary from platform to platform. E.g:
+                     * MacOS symlink delete and set (rm -rf operator && ln -s <path to>/file_settings/ operator):
+                     *     ENTRY_MODIFY:operator
+                     *     ENTRY_CREATE:settings.json
+                     *     ENTRY_MODIFY:settings.json
+                     * Linux in Docker symlink delete and set (rm -rf operator && ln -s <path to>/file_settings/ operator):
+                     *     ENTRY_CREATE:operator
+                     * After we get an indication that something has changed, we check the timestamp, file id,
+                     * real path of our desired file.
+                     */
+                    if (Files.exists(settingsDir)) {
+                        try {
+                            Path path = operatorSettingsFile();
+
+                            if (logger.isDebugEnabled()) {
+                                key.pollEvents().stream().forEach(e -> logger.debug("{}:{}", e.kind().toString(), e.context().toString()));
+                            }
+
+                            key.pollEvents();
+                            key.reset();
+
+                            enableSettingsWatcher(settingsDir);
+
+                            if (watchedFileChanged(path)) {
+                                processFileSettings(path, false);
+                            }
+                        } catch (Exception e) {
+                            logger.warn("unable to watch or read operator settings file", e);
+                        }
+                    } else {
+                        key.pollEvents();
+                        key.reset();
+                    }
+                }
+            } catch (Exception e) {
+                if (logger.isDebugEnabled()) {
+                    logger.debug("encountered exception watching", e);
+                }
+                logger.info("shutting down watcher thread");
+            } finally {
+                watcherThreadLatch.countDown();
+            }
+        }, "elasticsearch[file-settings-watcher]").start();
+    }
+
+    synchronized void stopWatcher() {
+        logger.debug("stopping watcher ...");
+        if (watching()) {
+            try {
+                watchService.close();
+                if (watcherThreadLatch != null) {
+                    watcherThreadLatch.await();
+                }
+            } catch (IOException | InterruptedException e) {
+                logger.info("encountered exception while closing watch service", e);
```

Comment:
why is this info instead of error or warn?

Reply:
I'll switch to warn.